simple_parsing/wrappers/field_wrapper.py

```diff
@@ -1042,3 +1022,62 @@ def get_argparse_options_for_annotation(t: Type) -> Dict[str, Any]:
     #     else:
     #         _arg_options["nargs"] = "*"
     return _arg_options
+
+
+def _optional_args(_arg_options, type_arguments):
+    _arg_options["required"] = False
+
+    # NOTE: Optional[<something>] is always translated to
+    # Union[<something>, NoneType]
+    assert type_arguments
+    non_none_types = [t for t in type_arguments if t is not type(None)]
+    assert non_none_types
+    if len(non_none_types) == 1:
+        wrapped_type = non_none_types[0]
+    else:
+        wrapped_type = Union[tuple(non_none_types)]
+    if utils.is_tuple(wrapped_type):
+        # TODO: ISSUE 42: better handle optional/nested tuples.
+        # For now we just assume that the item types are 'simple'.
+
+        # IDEA: This chould probably be a static method that takes in the type
+        # annotation, and uses a recursive call to fetch the arg options of the
+        # item type, and uses some of the entries from that dict to construct
+        # the arg options of the parent?
+        # NOTE: We'd have to return different value for the `type` argument
+        # depending on the nesting:
+        # Tuple[int, int] -> <class int>
+        # Tuple[int, str] -> <some autogenerated function>
+        # Tuple[Tuple[int, str], Tuple[int, str]] -> <some autogenerated
+        #   function *that doesn't just use `<class int> from above twice!
+        #   (Since we want to support passing --foo '(a, 1)' '(b, 4)'
+        # `)>
+        _arg_options["type"] = get_parsing_fn(wrapped_type)
+        _arg_options["nargs"] = utils.get_container_nargs(wrapped_type)
+
+    elif utils.is_list(wrapped_type):
+        _arg_options["type"] = utils.get_argparse_type_for_container(wrapped_type)
+        _arg_options["nargs"] = "*"
+        # NOTE: Can't set 'const', since we'd get:
+        # ValueError: nargs must be '?' to supply const
+        # _arg_options["const"] = []
+    else:
+        _arg_options["type"] = get_parsing_fn(wrapped_type)
+        # TODO: Should the 'nargs' really be '?' here?
+        _arg_options["nargs"] = argparse.OPTIONAL
+        # assert False, (wrapped_type, utils.is_tuple(wrapped_type))
+
+
+def _literal_args(_arg_options, t):
+    type_arguments = utils.get_type_arguments(t)
+
+    if len(set(type(arg) for arg in type_arguments)) != 1:
+        raise ValueError("Only Literals of the same type are supported")
+
+    literal_type = type(type_arguments[0])
+
+    if literal_type not in [str, int, bool, bytes] and utils.is_enum(literal_type):
```

Comment:
Wait, you're saying that enums are supported in the string below, but your test condition above uses `and utils.is_enum`, did you mean to write this instead?

```suggestion
    if literal_type not in [str, int, bool, bytes] and not utils.is_enum(literal_type):
```

And if so, then ideally some test should have been able to catch that!